src/Common/MoistThermodynamics/states.jl

```diff
@@ -176,28 +195,97 @@ end
 
 Constructs a [`PhaseNonEquil`](@ref) thermodynamic state from:
 
- - `θ_liq_ice` - liquid-ice potential temperature
- - `ρ` - density
- - `q_pt` - phase partition
+ - `θ_liq_ice` liquid-ice potential temperature
+ - `ρ` (moist-)air density
+ - `q_pt` phase partition
+and, optionally
+ - `tol` tolerance for non-linear equation solve
+ - `maxiter` maximum iterations for non-linear equation solve
 """
-function LiquidIcePotTempSHumNonEquil(θ_liq_ice::FT, ρ::FT, q_pt::PhasePartition{FT}) where {FT<:Real}
-    T = air_temperature_from_liquid_ice_pottemp_non_linear(θ_liq_ice, ρ, q_pt)
+function LiquidIcePotTempSHumNonEquil(θ_liq_ice::FT,
+                                      ρ::FT,
+                                      q_pt::PhasePartition{FT},
+                                      tol::FT=FT(1e-2),
+                                      maxiter::Int=10
+                                      ) where {FT<:Real}
+    T = air_temperature_from_liquid_ice_pottemp_non_linear(θ_liq_ice, ρ, tol, maxiter, q_pt)
     e_int = internal_energy(T, q_pt)
-    return PhaseNonEquil(e_int, ρ, q_pt)
+    return PhaseNonEquil{FT}(e_int, ρ, q_pt)
 end
 
 """
     LiquidIcePotTempSHumNonEquil_given_pressure(θ_liq_ice, p, q_pt)
 
 Constructs a [`PhaseNonEquil`](@ref) thermodynamic state from:
 
- - `θ_liq_ice` - liquid-ice potential temperature
- - `p` - pressure
- - `q_pt` - phase partition
+ - `θ_liq_ice` liquid-ice potential temperature
+ - `p` pressure
+ - `q_pt` phase partition
 """
 function LiquidIcePotTempSHumNonEquil_given_pressure(θ_liq_ice::FT, p::FT, q_pt::PhasePartition{FT}) where {FT<:Real}
     T = air_temperature_from_liquid_ice_pottemp_given_pressure(θ_liq_ice, p, q_pt)
     ρ = air_density(T, p, q_pt)
     e_int = internal_energy(T, q_pt)
-    return PhaseNonEquil(e_int, ρ, q_pt)
+    return PhaseNonEquil{FT}(e_int, ρ, q_pt)
+end
+
+"""
+    fixed_lapse_rate_ref_state(z::FT,
+                               T_surface::FT,
+                               T_min::FT) where {FT<:AbstractFloat}
+
+Fixed lapse rate hydrostatic reference state
+"""
+function fixed_lapse_rate_ref_state(z::FT,
```

Comment:
Any reason not to call the `ref_state.jl` code here?

Reply:
That requires arguments `m::HydrostaticState{P,F}, atmos::AtmosModel, aux::Vars, geom::LocalGeometry`, which is a bit higher level than the primitives we're dealing with here. Maybe the calls should be the other way around though, to avoid code duplication.

Reply:
Ok